CatalogReaderExample/NuGet.Protocol.Catalog/ICatalogLeafProcessor.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace NuGet.Protocol.Catalog
+{
+    /// <summary>
+    /// An interface which allows custom processing of catalog leaves. This interface should be implemented when the
+    /// catalog leaf documents need to be downloaded and processed in chronological order.
+    /// </summary>
+    public interface ICatalogLeafProcessor
+    {
+        /// <summary>
+        /// Process a catalog leaf containing package details. This method should return false or throw an exception
+        /// if the catalog leaf cannot be processed. In this case, the <see cref="CatalogProcessor" /> will stop
+        /// processing items.
```

Comment:
Should there be a note that there are no guarantees that a `ProcessPackageDetailsAsync` will be called per specifc package id/version? I feel that this may be a common incorrect assumption that consumers of this API will run into.

Reply:
Yeah, great point. I think this would be the most common gotcha. I'll add it.

Reply:
Fixed.